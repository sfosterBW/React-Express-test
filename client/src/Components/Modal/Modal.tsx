import React, { FC } from 'react'
import { RootState } from '../../Reducers/store'
import { useSelector, useDispatch } from 'react-redux'
import { toggleModal } from '../../Reducers/modalReducer'
import { RootActions } from '../../Reducers/store'
import { Fruit } from '../../utils/interfaces'
import EditForm from '../Form/EditForm'
import NewForm from '../Form/NewForm'
import styles from './Modal.module.css'

const Modal: FC<{fruit: Fruit | undefined}> = ({ fruit }) => {
  const dispatch = useDispatch()

  const toggle = useSelector((state: RootState) => state.modal.toggle)

  if(!toggle || !fruit) {
    return null
  }

  return (
    <div>
      <section className={styles.wrapper}>
        <div className={styles.modal}>
          {!fruit && <NewForm />}
          {fruit && <EditForm fruit={fruit} title="Edit fruit" />}
          <button
            className={styles.close}
            data-testid="close"
            name="close"
            onClick={(): RootActions => dispatch(toggleModal(!toggle))}
          >
            &times;
          </button>
        </div>
      </section>
    </div>
  )
}

export default Modal
