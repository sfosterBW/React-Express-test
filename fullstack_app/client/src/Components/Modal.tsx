import React, { FC } from 'react'
import { RootState } from '../utils/store'
import { useSelector, useDispatch } from 'react-redux'
import { toggleModal } from '../utils/actions'
import Form from './Form'
import { IFruit } from '../utils/interfaces'

interface Props {
  fruit: IFruit | undefined
  title: string
}

const Modal: FC<Props> = ({ fruit, title }) => {

  const selectToggle = (state: RootState) => state.modal.toggle
  const toggle = useSelector(selectToggle)
  const dispatch = useDispatch()

  const displayModal = () => {
    if (toggle && fruit) {
      return (
        <div className="modal-wrapper">
          <div className="modal">
            <h2 className="modal-title">{title}</h2>
            <Form fruit={fruit} />
            <button
              className="closeButton"
              name="close"
              onClick={() => dispatch(toggleModal(!toggle))}>
              &times;
              </button>
          </div>
        </div>
      )
    } else {
      console.log("Modal fruit undefined")
      return null
    }
  }

  return (
    <div>{displayModal()}</div>
  )
}

export default Modal
