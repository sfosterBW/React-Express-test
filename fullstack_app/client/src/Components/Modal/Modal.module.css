.wrapper {
  position: absolute;
  bottom: 0;
  right: 0;
  left: 0;
  top: 0;
  display: flex;
  justify-content: center;
  align-items: center;
}

.modal {
  position: relative;
  border-radius: 5px;
  background: lightgray;
}

.modal form {
  background: lightgray;
}

.title {
  font-size: 3vmin;
  text-align: center;
}

.close {
  position: absolute;
  top: 1.5vmin;
  right: 0px;
  transform: translateY(-50%);
  cursor: pointer;
}
