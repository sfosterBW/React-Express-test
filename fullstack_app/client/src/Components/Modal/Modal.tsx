import React, { FC } from 'react'
import { RootState } from '../../utils/store'
import { useSelector, useDispatch } from 'react-redux'
import { toggleModal } from '../../utils/actions'
import { IFruit } from '../../utils/interfaces'
import Form from '../Form/Form'
import styles from './Modal.module.css'

interface Props {
  fruit: IFruit | undefined
  title: string
}

const Modal: FC<Props> = ({ fruit, title }) => {

  const selectToggle = (state: RootState) => state.modal.toggle
  const toggle = useSelector(selectToggle)
  const dispatch = useDispatch()

  const displayModal = () => {
    if (toggle && fruit) {
      return (
        <section className={styles.wrapper}>
          <div className={styles.modal}>
            <h2 className={styles.title}>{title}</h2>
            <Form fruit={fruit} />
            <button
              className={styles.close}
              name="close"
              onClick={() => dispatch(toggleModal(!toggle))}>
              &times;
              </button>
          </div>
        </section>
      )
    }
  }

  return (
    <div>{displayModal()}</div>
  )
}

export default Modal
