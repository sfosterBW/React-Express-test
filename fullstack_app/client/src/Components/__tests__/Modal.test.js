
import React from 'react'
import { mount, render, shallow } from 'enzyme'
import renderer from 'react-test-renderer'

import Modal from '../Modal'

describe('the Modal component', () => {
  it('shallow renders', () => {
    const modal = mount(<Modal />)
    expect(modal).toBeDefined()
  })

  it('renders the same as last time', () => {
    const tree = renderer
      .create(<Modal />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
