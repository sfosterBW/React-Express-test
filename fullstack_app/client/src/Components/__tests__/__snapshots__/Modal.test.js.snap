// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Modal component renders the same as last time 1`] = `
<div
  className="modal-wrapper"
>
  <div
    className="modal"
  >
    <h2 />
    <button
      className="closeButton"
      name="close"
      onClick={[Function]}
    >
      Ã—
    </button>
  </div>
</div>
`;
